app/Http/Controllers/UsersController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Rogue\Http\Controllers;
+
+use Rogue\Models\Post;
+use Rogue\Models\Signup;
+use Rogue\Services\Fastly;
+use Rogue\Services\ImageStorage;
+
+class UsersController extends Controller
+{
+    /**
+     * Create a controller instance.
+     */
+    public function __construct(ImageStorage $storage, Fastly $fastly)
+    {
+        $this->storage = $storage;
+        $this->fastly = $fastly;
+
+        $this->middleware('auth:api');
+        $this->middleware('role:admin,staff');
+        $this->middleware('scopes:write');
+        $this->middleware('scopes:activity');
+    }
+
+    /**
+     * Delete the given user's activity.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        // Find the user's signups, anonymize, and soft-delete.
+        $signups = Signup::withTrashed()->where('northstar_id', $id);
+        foreach ($signups->cursor() as $signup) {
+            $signup->update([
+                'why_participated' => null,
+                'details' => null,
+            ]);
```

Comment:
Sorry, I know this is not new functionality, but out of curiosity -- is the idea that we purge the user's signups & posts of data & _soft_ delete so that we still retain their count for our total impact numbers?

Reply:
Yes! The (totally undocumented) intent here is that we want to keep as much non-identifiable information about deleted activity as we can so that we could infer trends down the line. (This came out of an audit that Megha & I did when initially building this out to see which fields could possibly include PII & what had anonymized analytic value).

Reply:
I imagine that soft-deleted records would _not_ count for impact in Quasar (but I'm not 100% sure on that).

Reply:
Ah ok, that makes more sense, thanks for clarifying!